library/src/main/java/SpecTreeToYamlStringConverter.java

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Provides the ability to serialize Spec Trees represented as LinkedHashMaps into YAML strings */
+public class SpecTreeToYamlStringConverter {
+
+  int indent;
+
+  public SpecTreeToYamlStringConverter() {
+    this.indent = 2;
```

Comment:
change this to class level constant (SPACES_TO_INDENT) to make it more readable

Reply:
The reason I made it this way was to potentially add a new constructor in the future to specify a custom indent. But I also agree with making it a constant for now as I cant foresee a need for that.